ledfx/api/virtual_tools.py

```diff
@@ -61,8 +62,8 @@ async def put(self, request: web.Request) -> web.Response:
             for virtual_id in self._ledfx.virtuals:
                 virtual = self._ledfx.virtuals.get(virtual_id)
                 if virtual is not None:
-                    virtual.oneshot(
-                        parse_color(validate_color(color)), ramp, hold, fade
+                    virtual.add_oneshot(
```

Comment:
I do like that you are improving the code by moving things like validate closer to the color substantiation, makes me happy!